### PR TITLE
Add basic reverseproxy misconfig detection to setupchecks

### PR DESCRIPTION
**Pull request by rullzer**
This seesms to be a common cause of error for a lot of people behind a reverse proxy.

TODO:
- [x] Tests
- [ ] Backports

### CODE REVIEW

**Comment by kesselb** [comment]
@rullzer fyi: https://github.com/nextcloud/docker/pull/819

File: core/js/setupchecks.js [hunk 1]
```diff
@@ -460,6 +460,18 @@
 							type: OC.SetupChecks.MESSAGE_TYPE_WARNING
 						})
 					}
+					if (window.location.protocol === 'http:' && data.reverseProxyGeneratedURL.split('/')[0] !== 'https:') {
```
Comment (ChristophWurst) [inline core/js/setupchecks.js]: alternatively you may use `.startsWith` https://www.w3schools.com/jsref/jsref_startswith.asp

**Review by ChristophWurst — APPROVED** [review]
Looks good!

File: core/js/setupchecks.js (hunk 1 above)
Comment (juliushaertl) [reply core/js/setupchecks.js]: I think the file is not bundled yet, so that will probably not work with IE11.

**Review by juliusknorr — APPROVED** [review]

File: core/js/setupchecks.js (hunk 1 above)
Comment (ChristophWurst) [reply core/js/setupchecks.js]: Yes, good point. Nevermind then

File: core/js/setupchecks.js [hunk 2]
```diff
@@ -460,6 +460,18 @@
...
+						messages.push({
+							msg: t(
+								'core',
+								'You are accessing your instance over a secure connection, however your instance is generating insecure URLs. This most likely means that you are behind a reverse proxy and the overwrite config variables are not set correctly. Please read <a target="_blank" rel="noreferrer noopener" href="{docLink}">the documentation page about this</a>.',
```
Comment (nickvergessen) [inline core/js/setupchecks.js]: Please don't add HTML in translations.
Read https://docs.nextcloud.com//server/stable/developer_manual/app/view/l10n.html#improving-your-translations as a penalty and send a follow up PR, thanks :)

**Pull request merged**